<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [pentf](./pentf.md) &gt; [expectedToFail](./pentf.expectedtofail.md)

## expectedToFail() function

Mark a code section as expected to fail. If the async function throws an error, the error will be included in reports, but not counted as a test failure. If the async function succeeds, a warning will be printed.

<b>Signature:</b>

```typescript
export declare function expectedToFail(config: Config, message: string, asyncFunc: () => any, { expectNothing }?: {
    expectNothing?: boolean | undefined;
}): Promise<void>;
```

## Parameters

|  Parameter | Type | Description |
|  --- | --- | --- |
|  config | Config | The pentf configuration. |
|  message | string | Error message to show when the section fails (recommended: ticket URL) |
|  asyncFunc | () =&gt; any | The asynchronous section which is part of the test. |
|  { expectNothing } | { expectNothing?: boolean \| undefined; } |  |

<b>Returns:</b>

Promise&lt;void&gt;

## Example


```
await expectedToFail(config, 'BUG-1234', async() => {
    ...
}, {
    expectNothing: config.env === 'very-good-environment',
});

```

